File: common/src/main/java/cn/breadnicecat/candycraftce/recipe/RecipeSerializerExt.java
package cn.breadnicecat.candycraftce.recipe;

import com.google.gson.JsonObject;
import net.minecraft.world.item.crafting.Recipe;
import net.minecraft.world.item.crafting.RecipeSerializer;

public interface RecipeSerializerExt<R extends Recipe<?>> extends RecipeSerializer<R> {
	void toJson(JsonObject object, R recipe);

}

File: src/main/kotlin/cn/breadnicecat/candycraftce/core/recipe/RecipeSerializerExt.kt
package cn.breadnicecat.candycraftce.core.recipe

import com.google.gson.JsonObject
import net.minecraft.world.item.crafting.Recipe
import net.minecraft.world.item.crafting.RecipeSerializer

interface RecipeSerializerExt<R : Recipe<*>> : RecipeSerializer<R> {
    fun toJson(json: JsonObject, recipe: R)
}
